### PR TITLE
ORLP-432 (Refactoring after investigating DB and Entyties)

### PR DESCRIPTION
**Pull request by serhiivenhrynovych**

### CODE REVIEW

File: src/main/java/com/softserve/academy/spaced/repetition/domain/Account.java [hunk 1]
```diff
@@ -27,18 +27,18 @@
     @Column(name = "account_id", nullable = false)
     private Long id;
 
-    @Column(name = "password")
+    @Column(name = "password", length = PASS_MAX_SIZE)
     @Size(message = PASS_SIZE_MESSAGE, min = PASS_MIN_SIZE, max = PASS_MAX_SIZE, groups = Request.class)
     private String password;
 
-    @Column(name = "email", unique = true, nullable = false)
+    @Column(name = "email", unique = true, nullable = false, length = EMAIL_MAX_SIZE)
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/domain/Account.java]: Is it necessary to set nullable = false, in annotation Column, if we have annotation NotNull below?

File: src/main/java/com/softserve/academy/spaced/repetition/domain/Account.java [hunk 2]
```diff
@@ -27,18 +27,18 @@
...
     @NotNull(message = NULL_MESSAGE, groups = Request.class)
     @Size(min = EMAIL_MIN_SIZE, max = EMAIL_MAX_SIZE, message = EMAIL_SIZE_MESSAGE, groups = Request.class)
     @Pattern(regexp = EMAIL_PATTERN, message = EMAIL_PATTERN_MESSAGE, groups = Request.class)
     @EmailNotExist(groups = Request.class)
     private String email;
 
-    @Column(name = "AUTHENTICATIONTYPE", length = 8)
+    @Column(name = "authenticationtype", length = AUTH_TYPE_MAX_SIZE)
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/domain/Account.java]: Maybe better to write `authentication_type` (not sure)

File: src/main/java/com/softserve/academy/spaced/repetition/domain/Account.java [hunk 3]
```diff
@@ -52,7 +52,7 @@
     @NotNull
     private boolean deactivated;
 
-    @Column(name = "LASTPASSWORDRESETDATE")
+    @Column(name = "lastpasswordresetdate")
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/domain/Account.java]: The same `last_password_reset_date`

File: src/main/java/com/softserve/academy/spaced/repetition/domain/Category.java [hunk 4]
```diff
@@ -6,7 +6,7 @@
 import java.util.List;
 
 @Entity
-@Table(name = "Category")
+@Table(name = "category")
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/domain/Category.java]: Why did you write `category` in lower case?

Comment (MoGUREV) [reply src/main/java/com/softserve/academy/spaced/repetition/domain/Category.java]: Okay, sorry, I understand)

File: src/main/java/com/softserve/academy/spaced/repetition/utils/validators/ValidationConstants.java [hunk 5]
```diff
@@ -26,7 +26,7 @@ private ValidationConstants() {
 
     public static final String SPECIAL_SYMBOLS_PATTERN_MESSAGE = "Can not consist of reserved symbols";
 
-    public static final int EMAIL_MIN_SIZE = 3;
+    public static final int EMAIL_MIN_SIZE = 6;
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/utils/validators/ValidationConstants.java]: Why 6?

File: src/main/resources/db/migration/V1.0__Create_initial_schema.sql [hunk 6]
```diff
@@ -1,12 +1,12 @@
 create table account (
   account_id bigint not null auto_increment,
-  authenticationtype varchar(8) not null,
+  authentication_type varchar(8) not null,
```
Comment (MoGUREV) [inline src/main/resources/db/migration/V1.0__Create_initial_schema.sql]: We have a migration tool, so, maybe better to create migration script.

File: src/main/java/com/softserve/academy/spaced/repetition/domain/Account.java [hunk 7]
```diff
@@ -52,9 +53,9 @@
     @NotNull
     private boolean deactivated;
 
-    @Column(name = "LASTPASSWORDRESETDATE")
-    @Temporal(TemporalType.TIMESTAMP)
+    @Column(name = "last_password_reset_date")
     @NotNull
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/domain/Account.java]: Why did you add NotNull to this field?

Comment (MoGUREV) [reply src/main/java/com/softserve/academy/spaced/repetition/domain/Account.java]: Ou, sorry))

File: src/main/java/com/softserve/academy/spaced/repetition/domain/Image.java [hunk 8]
```diff
@@ -13,17 +14,19 @@
     @Column(name = "image_id")
     private Long id;
 
-    @Column(name = "imagebase64", columnDefinition = "LONGTEXT")
+    @Column(name = "image_base64", columnDefinition = "LONGTEXT")
     private String imagebase64;
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/domain/Image.java]: maybe better to write `imageBase64`, but not sure

File: src/main/java/com/softserve/academy/spaced/repetition/utils/validators/ValidationConstants.java [hunk 9]
```diff
@@ -48,6 +48,14 @@ private ValidationConstants() {
 
     public static final int PASS_MAX_SIZE = 20;
 
+    public static final int PASS_MAX_SIZE_VAL = 60;
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/utils/validators/ValidationConstants.java]: We have max limit for password above `PASS_MAX_SIZE = 20`

File: src/main/java/com/softserve/academy/spaced/repetition/utils/validators/ValidationConstants.java [hunk 10]
```diff
@@ -48,6 +48,14 @@ private ValidationConstants() {
...
+
+    public static final int AUTH_TYPE_MAX_SIZE = 8;
```
Comment (MoGUREV) [inline src/main/java/com/softserve/academy/spaced/repetition/utils/validators/ValidationConstants.java]: maybe better to write full name like `AUTHENTICATION_TYPE_MAX_SIZE` and so on

File: src/main/java/com/softserve/academy/spaced/repetition/utils/validators/ValidationConstants.java (hunk 9 above)
Comment (serhiivenhrynovych) [reply src/main/java/com/softserve/academy/spaced/repetition/utils/validators/ValidationConstants.java]: we need different constant for DB because passwordEncoder  hashes password to 60 characters, and normal password should be 20
